syntax = "proto3";

message GamesData {
	string id = 1;
	string name = 2;
    int32 reviews_number = 3;
	
}

message GetMostReviewedGamesRequest {
    int32 max_results = 1;
}

message GamesDataResponse {
    repeated GamesData game = 1;
}


message GameByIdRequest {
    string id = 1;
}


message GameByNameRequest {
    string name = 1;
}


service Games {
    rpc GetGames (GetMostReviewedGamesRequest) returns (GamesDataResponse);
    rpc GameByID (GameByIdRequest) returns (GamesData);
	rpc GameByName (GameByNameRequest) returns (GamesData);

}